### DISCUSSION TITLE
Enabling Single Writer Multiple Reader (SWMR)

### DISCUSSION THREAD

**Discussion by amcassidy** [aldanor/hdf5-rust#205] [Questions]
Hi,

I'm relatively new to Rust and HDF5.   What is the simplest way to enable SWMR when creating a file like the following:

 let mut file = File::create(data_file_name.clone())?; 

Thanks,

Andrew

**Comment by mulimoen** [comment]
This is not included in the library yet, but you could call the `hdf5-sys` function,
```rust
let err = hdf5_sys::h5f::H5Fstart_swmr_write(file.id());
```
You might have to bump the `libver` of the file, swmr is not supported in e.g. 1.8

**Comment by amcassidy** [comment]
Hi mulimoen,
Thank you for the reply.  

I'm currently using hdf5 version: 1.12.1  I've added to the code as suggested:

```
    let mut file = File::create(data_file_name.clone())?; // open for writing    
    let err = unsafe{h5f::H5Fstart_swmr_write(file.id())};
    println!("Error: {}", err);
```

hdf5_sys::h5f::**hdf5_1_10_0**
pub unsafe fn H5Fstart_swmr_write(file_id: hid_t) -> herr_t
=>  Error: -1
Which should be 0 correct? I am not sure where I would bump the libver, can you you point it out?

Thank you.

**Reply by mulimoen** [reply]
The following code correctly returns 0:
```rust
use hdf5::*;

fn main() {
    let file = FileBuilder::new()
        .with_fapl(|fapl| {
            fapl.libver_bounds(file::LibraryVersion::V110, file::LibraryVersion::V110)
        })
        .create("file.h5")
        .unwrap(); // open for writing
    let err = unsafe { hdf5_sys::h5f::H5Fstart_swmr_write(file.id()) };
    println!("Error: {}", err);
}
```

**Comment by amcassidy** [comment]
Thank you mulimoen, that works.
